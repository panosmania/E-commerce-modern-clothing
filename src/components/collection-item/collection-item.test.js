import { shallow } from "enzyme";
import React from "react";
import toJson from "enzyme-to-json";

import { CollectionItem } from "./collection-item.component";

describe("CollectionItem", () => {
  let wrapper;
  let mockaddItem;
  const imageUrl = "www.testImage.com";
  const mockName = "black hat";
  const mockPrice = 10;

  beforeEach(() => {
    mockaddItem = jest.fn();

    const mockProps = {
      item: {
        imageUrl: "www.testImage.com",
        price: mockPrice,
        name: mockName,
      },
      addItem: mockaddItem,
    };

    wrapper = shallow(<CollectionItem {...mockProps} />);
  });

  it("should render CollectionItem component", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should call addItem when AddButton clicked  ", () => {
    wrapper.find(`[id='test']`).simulate("click");
    expect(mockaddItem).toHaveBeenCalled();
  });

  //κατι κανω λαθος εδω και δεν μου βγαζει αυτο που θελω
  //expect(wrapper.find(".image").prop("style")).toBe(imageUrl); λαθος δεν μπορει να γινει ετσι
  //με τον κατω τροπο δουλευει οπως πρεπει
  it("should render imageUrl as a prop on BackgroundImage", () => {
    expect(wrapper.find(".image").prop("style")).toHaveProperty(
      "backgroundImage",
      `url(${imageUrl})`
    );
  });

  it("should call addItem when AddButton clicked  ", () => {
    wrapper.find(`[id='test']`).simulate("click");
    expect(mockaddItem).toHaveBeenCalled();
  });

  it("should render name prop in NameContainer", () => {
    expect(wrapper.find(".name").text()).toBe(mockName);
  });

  it("should render price prop in PriceContainer", () => {
    const price = parseInt(wrapper.find(".price").text());
    expect(price).toBe(mockPrice);
  });
});
