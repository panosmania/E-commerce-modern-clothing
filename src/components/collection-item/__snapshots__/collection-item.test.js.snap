// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionItem should render CollectionItem component 1`] = `
<div
  className="collection-item"
>
  <div
    className="image"
    style={
      Object {
        "backgroundImage": "url(www.testImage.com)",
      }
    }
  />
  <div
    className="collection-footer"
  >
    <span
      className="name"
    >
      black hat
    </span>
    <span
      className="price"
    >
      10
    </span>
  </div>
  <CustomButton
    id="test"
    inverted={true}
    onClick={[Function]}
  >
    Add to cart
  </CustomButton>
</div>
`;
